src/vehicle.cpp

```diff
@@ -6488,31 +6466,12 @@ int vehicle::manual_fire_turret( int p, player &shooter, item &gun )
         // Put our shooter on the roof of the vehicle
         shooter.add_effect( effect_on_roof, 1 );
 
-        int to_fire = abs(parts[p].mode) > 1 ? std::max( gun.gun_get_mode( "AUTO" ).qty, 1 ) : 1;
-        res = shooter.fire_gun( targ, to_fire, gun );
+        int to_fire = abs(parts[p].mode) > 1 ? std::max( gun->gun_get_mode( "AUTO" ).qty, 1 ) : 1;
```

Comment:
`AUTO` should be the largest burst fire mode. I do agree though that the turret code should enumerate the various modes and that was one of the design criteria when I rewrote the gunmod code